javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,145 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
```

Comment:
Very impressed with your solution! Looks good and it works, and the code is easy to understand.

Your extra features are also well thought out.

Only one minor bug found: It I search for e.g. "carrot", then select "Clear Filters", it will show all products, as intended. If I then click either "Cheapest first" or "Best First", it will only show products with only "carrot" in it, even though the search bar is empty.

Reply:
I have tried getting rid of this bug in different ways while doing the homework but I haven't managed so far. Could you please tell me what the problem is with it? It might be either in the event clearing or in the filter function 🤔

Reply:
I’ll check it out and see if I can spot it :)

Reply:
Problem solved! Had to refactor my function to make a general one for both filter and sort